### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by mortonjt**

### CODE REVIEW

File: micronota/bfillings/diamond.py [hunk 1]
```diff
@@ -315,12 +330,105 @@ def parse_tabular(diamond_res, column='bitscore'):
...
+            file path
+        column : str
+            The column used to pick the best hits.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The best matched records for each query sequence.
+        '''
+        seqs = list(read(diamond_res, format='sam'))
```
Comment (RNAer) [inline micronota/bfillings/diamond.py]: can it be iterator instead of materialized list?

File: micronota/bfillings/diamond.py [hunk 2]
```diff
@@ -315,12 +330,105 @@ def parse_tabular(diamond_res, column='bitscore'):
...
+    Attributes
+    ----------
+    cache : list of skbio.Sequence
+        List of sequences to build cache from
+
+    '''
+    def __init__(self, seqs=None, maxSize=200000, out_dir=""):
+        self.fname = self._generate_random_file() # substitute for tempfile
+        self.fasta = join(out_dir, '%s.fasta' % self.fname)
+        self.db = join(out_dir, '%s.diamond.dmnd' % self.fname)
```
Comment (RNAer) [inline micronota/bfillings/diamond.py]: the naming standard is xxx.dmnd. what is rational using `diamond.dmnd`?

File: micronota/bfillings/diamond.py [hunk 3]
```diff
@@ -315,12 +330,105 @@ def parse_tabular(diamond_res, column='bitscore'):
...
+class DiamondCache():
+    '''
+    Attributes
+    ----------
+    cache : list of skbio.Sequence
+        List of sequences to build cache from
+
+    '''
+    def __init__(self, seqs=None, maxSize=200000, out_dir=""):
+        self.fname = self._generate_random_file() # substitute for tempfile
```
Comment (RNAer) [inline micronota/bfillings/diamond.py]: why is it better than `tempfile`?

Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Its not - it is worse.  That was just for prototyping.  Will change.

File: micronota/bfillings/diamond.py (hunk 2 above)
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Will change.

File: micronota/bfillings/diamond.py (hunk 1 above)
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Done.

File: micronota/bfillings/diamond.py (hunk 3 above)
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Here's why I did this before: http://stackoverflow.com/questions/15169101/how-to-create-a-temporary-file-that-can-be-read-by-a-subprocess

I was running into some issues with tempFile.

Comment (mortonjt) [inline micronota/bfillings/diamond.py]: So I don't think this can be fixed with tempfile.

**Comment by mortonjt** [comment]
Fixed most of the stylistic changes.  However, this will fail, because it is importing sam.

BUT - `sam` is actually required for the parsers to work.  Not sure how to get travis to ignore this.

**Comment by RNAer** [comment]
you have to import it globally to make it work. see embl as an example:
https://github.com/biocore/micronota/blob/master/micronota/__init__.py

**Comment by mortonjt** [comment]
:fireworks: tests passing.  Ready for review.

**Comment by mortonjt** [comment]
Thanks!

**Pull request merged**